qutip/tests/test_qubitcircuit.py

```diff
@@ -203,31 +203,88 @@ def test_add_state(self):
         qc.add_state("-", targets=[1])
 
         assert_(qc.input_states[0] == "0")
-        assert_(qc.input_states[2] == None)
+        assert_(qc.input_states[2] is None)
         assert_(qc.output_states[1] == "+")
 
         qc1 = QubitCircuit(10)
 
         qc1.add_state("0", targets=[2, 3, 5, 6])
-        qc1.add_state("+", targets=[1,4,9])
-        qc1.add_state("A", targets=[1,4,9], state_type="output")
-        qc1.add_state("A", targets=[1,4,9], state_type="output")
+        qc1.add_state("+", targets=[1, 4, 9])
+        qc1.add_state("A", targets=[1, 4, 9], state_type="output")
+        qc1.add_state("A", targets=[1, 4, 9], state_type="output")
         qc1.add_state("beta", targets=[0], state_type="output")
-        assert_(qc1.input_states[0] == None)
+        assert_(qc1.input_states[0] is None)
 
         assert_(qc1.input_states[2] == "0")
         assert_(qc1.input_states[3] == "0")
         assert_(qc1.input_states[6] == "0")
         assert_(qc1.input_states[1] == "+")
         assert_(qc1.input_states[4] == "+")
 
-        assert_(qc1.output_states[2] == None)
+        assert_(qc1.output_states[2] is None)
```

Comment:
_If_ we were to keep using the `np.testing` functions (which we don't need to because `pytest` makes it all easier), the better one here would be `np.testing.assertEqual` (even for the `None` test).

We always should think about what information we want to get out of a failed test.  In this case, we'd be best served by knowing what the incorrect value was, so we should pass the two parts of the comparison separately.  That way, `np.testing.assertEqual` will print them out on a test failure, whereas this way `np.testing.assert_` just gets fed the value `False` and we wouldn't know what it was (was it 3, 0, "", "None"?) and we'd have to do more work offline to find out.